examples/atmega2560/src/bin/atmega2560-i2cdetect.rs

```diff
@@ -0,0 +1,38 @@
+#![no_std]
+#![no_main]
+
+use atmega_hal::clock;
+use atmega_hal::usart::{Baudrate, Usart};
+use panic_halt as _;
+
+type I2c = atmega_hal::i2c::I2c<clock::MHz16>;
+
+#[avr_device::entry]
+fn main() -> ! {
+    let dp = atmega_hal::Peripherals::take().unwrap();
+    let pins = atmega_hal::pins!(dp);
+
+    // set up serial interface for text output
+    let mut serial = Usart::new(
+        dp.USART0,
+        pins.pe0,
+        pins.pe1.into_output(),
+        Baudrate::<clock::MHz16>::new(57600),
+    );
```

Comment:
Ah, not entirely correct, I do have a comment ;)

You're hard coding the clock-rate in different parts of the code here.  This works, but it will make changes very tedious later on.  Instead, I would recommend to define a project-wide clock type alias which is then referenced everywhere else.  Maybe you can do this in the examples to demonstrate what people should be doing?

In code, it would look like this:

```rust
/// The project-global CPU clock speed.  This type alias should be defined once
/// in a project and then referenced whereever types need the clock speed.
type CoreClock = atmega_hal::clocl::MHz16;

type I2c = atmega_hal::i2c::I2c<crate::CoreClock>;

// ...

    let mut serial = Usart::new(
        dp.USART0,
        pins.pe0,
        pins.pe1.into_output(),
        Baudrate::<crate::CoreClock>::new(57600),
    );
```

Reply:
That's a good point, thanks!